pkg/apis/config/v1alpha1/meshrootcertificate.go

```diff
@@ -0,0 +1,106 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// MeshRootCertificate defines the configuration for certificate issuing
+// by the mesh control plane
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type MeshRootCertificate struct {
+	// Object's type metadata
+	metav1.TypeMeta `json:",inline"`
+
+	// Object's metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec is the MeshRootCertificate config specification
+	// +optional
+	Spec MeshRootCertificateSpec `json:"spec,omitempty"`
+
+	// Status of the MeshRootCertificate resource
+	// +optional
+	Status MeshRootCertificateStatus `json:"status,omitempty"`
+}
+
+// MeshRootCertificateSpec defines the mesh root certificate specification
+type MeshRootCertificateSpec struct {
+	// Provider specifies the mesh certificate provider
+	// +optional
+	Provider *ProviderSpec `json:"provider,omitempty"`
```

Comment:
Is there a reason the only field in the spec is optional? Does an empty spec mean something?